src/credential/sd_jwt_vc.rs

```diff
@@ -0,0 +1,169 @@
+    payload_json_string: String,
+    disclosures: Option<Vec<String>>,
+    key_alias: Option<KeyAlias>,
+}
+
+#[uniffi::export]
+impl SdJwtVc {
+    /// The VdcCollection ID for this credential.
+    pub fn id(&self) -> Uuid {
+        self.id
+    }
+
+    /// The version of the Verifiable Credential Data Model that this credential conforms to.
+    pub fn vcdm_version(&self) -> VcdmVersion {
+        match &self.credential {
+            ssi::claims::vc::AnySpecializedJsonCredential::V1(_) => VcdmVersion::V1,
+            ssi::claims::vc::AnySpecializedJsonCredential::V2(_) => VcdmVersion::V2,
+        }
+    }
+
+    /// The type of this credential. Note that if there is more than one type (i.e. `types()`
+    /// returns more than one value), then the types will be concatenated with a "+".
+    pub fn r#type(&self) -> CredentialType {
+        CredentialType(self.types().join("+"))
+    }
+
+    /// The types of the credential from the VCDM, excluding the base `VerifiableCredential` type.
+    pub fn types(&self) -> Vec<String> {
+        match &self.credential {
+            ssi::claims::vc::AnySpecializedJsonCredential::V1(vc) => vc.additional_types().to_vec(),
+            ssi::claims::vc::AnySpecializedJsonCredential::V2(vc) => vc.additional_types().to_vec(),
+        }
+    }
+
+    /// Access the W3C VCDM credential as a JSON encoded UTF-8 string.
+    pub fn credential_as_json_encoded_utf8_string(&self) -> String {
+        self.credential_string.clone()
+    }
+
+    /// Access the JWS header as a JSON encoded UTF-8 string.
+    pub fn sd_jwt_header_as_json_encoded_utf8_string(&self) -> String {
+        self.header_json_string.clone()
+    }
+
+    /// Access the JWS payload as a JSON encoded UTF-8 string.
+    pub fn sd_jwt_payload_as_json_encoded_utf8_string(&self) -> String {
+        self.payload_json_string.clone()
+    }
+
+    /// Access the revealed SD-JWT disclosures as a JSON encoded UTF-8 string.
+    pub fn sd_jwt_disclosures_as_json_encoded_utf8_string(&self) -> Option<String> {
+        self.disclosures
+            .as_ref()
+            .map(|disclosures| serde_json::to_string(disclosures).unwrap())
+    }
+
+    /// The keypair identified in the credential for use in a verifiable presentation.
+    pub fn key_alias(&self) -> Option<KeyAlias> {
+        self.key_alias.clone()
+    }
+}
+
+#[uniffi::export]
+impl SdJwtVc {
+    pub(crate) fn to_compact_sd_jwt_bytes(&self) -> Vec<u8> {
+        self.sd_jwt.as_bytes().to_vec()
+    }
+}
+
+#[uniffi::export]
+pub fn decode_reveal_sd_jwt(input: String) -> Result<String, SdJwtVcInitError> {
+    let jwt: SdJwtBuf = SdJwtBuf::new(input).map_err(|_| SdJwtVcInitError::InvalidSdJwt)?;
+    let revealed_jwt: RevealedSdJwt<AnyClaims> = jwt
+        .decode_reveal_any()
+        .map_err(|_| SdJwtVcInitError::JwtDecoding)?;
+    let claims: &JWTClaims = revealed_jwt.claims();
+    serde_json::to_string(claims).map_err(|_| SdJwtVcInitError::Serialization)
```

Comment:
There is no reason for the serialization to fail. `JWTClaim` is compatible with the JSON datamodel, you can safely `unwrap` and get rid of the `SdJwtVcInitError::Serialization` error variant.

Reply:
I think this okay to leave as-is, even though the `.map_err` will never be triggered, to your point, that we're already guaranteed to have a valid `JWTClaims` object that can be serialized.

Will remove if it is a blocker.